distributed/sizeof.py

```diff
@@ -62,3 +62,18 @@ def sizeof_pandas_index(i):
         p = int(i.memory_usage())
         obj = len(i) * 100 if i.dtype == object else 0
         return int(p + obj) + 1000
+
+
+@sizeof.register_lazy("scipy")
+def register_spmatrix():
+    from scipy import sparse
+    @sizeof.register(sparse.dok_matrix)
+    def sizeof_spmatrix_dok(s):
+        return s.__sizeof__()
+
+    @sizeof.register(sparse.spmatrix)
+    def sizeof_spmatrix(s):
+        return sum(
+            sizeof(v) for v in s.__dict__.values()
+        )
```

Comment:
It's very fast, just a couple milliseconds,, as the number of elements in the dictionary around 5 or 6 in each `spmatrix` and 2 or 3 are `ndarray`s depending on the sparse matrix type.
If this is a problem, an alternative would be to overestimate it as 
`len(s.__dict__) * int(s.data.nbytes)`
as `s.data` is a numpy ndarray in all cases except in  the `dok_matrix` type